### PR TITLE
Fix for getaddrinfo failures on Windows 10 in MongoDbContainer

### PR DESCRIPTION
**Pull request by arpost**
Windows npipe URIs are rewritten by the docker-py client library to http+docker://localnpipe. Other pipe-like URIs are rewritten this way too. So, `core/docker_client.py` should detect the http+docker scheme instead of "npipe" and return localhost. Because the check for "http" schemes will inadvertently pick up the http+docker scheme too, do the check for pipes first and for real URLs second. Fixes issue #99. All tests pass with this fix on Windows 10 with Docker Desktop running locally.

### CODE REVIEW

File: testcontainers/core/docker_client.py [hunk 1]
```diff
@@ -62,11 +62,12 @@ def host(self):
 
         except ValueError:
             return None
-        if 'http' in url.scheme or 'tcp' in url.scheme:
-            return url.hostname
-        if 'unix' in url.scheme or 'npipe' in url.scheme:
+        if 'unix' in url.scheme or 'http+docker' == url.scheme:
```
Comment (tillahoffmann) [inline testcontainers/core/docker_client.py:65]: Should `npipe` also be included in this check for backwards compatibility?

**Comment by arpost** [comment]
That’s a good idea. I don’t know when the docker client started replacing npipe with the special URL.

**Comment by arpost** [comment]
I updated the pull request with the npipe check.

File: tests/test_nginx_container.py [hunk 2]
```diff
@@ -9,6 +10,7 @@ def test_docker_run_nginx():
         port = nginx.port_to_expose
         url = "http://{}:{}/".format(nginx.get_container_host_ip(),
                                      nginx.get_exposed_port(port))
+        time.sleep(5)
```
Comment (tillahoffmann) [inline tests/test_nginx_container.py:13]: Can we drop the sleeps?

**Comment by arpost** [comment]
I have those in my copy because the tests fail on my hardware without them, but I didn’t mean for them to end up in my pull request. Sorry about that, I’ll move them so they aren’t in the pull request.

**Comment by alexanderankin** [comment]
what was the end outcome here?

**Pull request closed without merging**